pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -6519,15 +6547,25 @@ func TestReconcilePipeline_FinalTasks(t *testing.T) {
 			),
 		},
 
-		expectedTaskRuns: map[string]*v1beta1.PipelineRunTaskRunStatus{
-			"task-run-dag-task-1": getTaskRunStatus("dag-task-1", corev1.ConditionUnknown),
-		},
+		expectedChildReferences: []v1beta1.ChildStatusReference{{
+			TypeMeta: runtime.TypeMeta{
+				APIVersion: v1beta1.SchemeGroupVersion.String(),
+				Kind:       "TaskRun",
+			},
+			Name:             "task-run-dag-task-1",
+			PipelineTaskName: "dag-task-1",
+		}},
 
 		pipelineRunStatusUnknown: true,
 	}}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			// TODO: Refactor getTaskRun to populate matched OwnerReference with PipelineRun as in https://github.com/tektoncd/pipeline/issues/6008
```

Comment:
IMHO, we could create a helper func called `withOwnerReference(tr *v1beta1.TaskRun)` to avoid this workaround which looks kind of ugly 😅. That helper func can be removed in the `getTaskRun` refactoring PR. wdyt?

Reply:
Thanks @XinruZhang that makes sense. Updated